src/database/Connector.java

```diff
@@ -103,111 +180,119 @@ public synchronized boolean dropAllAccounts() {
-        // the account exists, check password
-        selectPassword.setString(1, username);
-        try (ResultSet resultPassword = selectPassword.executeQuery()) {
-          if (resultPassword.next()) {
-            // get the stored password from database
-            String passwordInDB = resultPassword.getString(1);
-            // password does not match
-            if (!passwordInDB.equals(password)) {
-              //System.err.println("the password does not match");
-              return false;
-            }
-            //System.out.println("the account exists");
-            return true;
+      try (PreparedStatement selectSalt = conn.prepareStatement(SQL_SELECT_SALT)) {
+        selectSalt.setString(1, username);
+        try (ResultSet resultSalt = selectSalt.executeQuery()) {
+          if (resultSalt.next()) {
+            // Get the stored password from database
+            return resultSalt.getString(1);
           }
         }
       }
-    } catch (SQLException e) {e.printStackTrace();return false;}
-    //System.err.println("the account does not exist");
-    return false;
+    }
+    catch (SQLException e) {
+      e.printStackTrace();
+      return null;
+    }
+    return null;
   }
 
-  /**
-   * verify if the account username input by users matches an account in the
-   * database
-   *
-   * @param username
-   * @return true if the account exists, false if the account does not exist
+  /** Verify if the account username and password input by users match what has been recorded in
+   *  database
+   * 
+   * @param username the username that is being verified
+   * @param password the password that is used to compare to the one in database
+   * @return true if the account is verified, else, false, if the account is not valid
    */
-  public synchronized boolean verifyAccountExists(String username) {
+  public synchronized boolean verifyAccount(String username, String password) {
 
-    String sqlSelectPassword = "select password from " + tableName + " where username = ?";
     try (Connection conn = ds.getConnection()) {
-      try (PreparedStatement selectPassword = conn.prepareStatement(sqlSelectPassword)) {
-        // the account exists
+      try (PreparedStatement selectPassword = conn.prepareStatement(SQL_SELECT_PASSWORD)) {
+        // The account exists, check password
         selectPassword.setString(1, username);
         try (ResultSet resultPassword = selectPassword.executeQuery()) {
           if (resultPassword.next()) {
+            // Get the stored password from database
+            String passwordInDB = resultPassword.getString(1);
+
+            // Encrypt password:
+            String salt = acquireSalt(username);
+            String codedPassword = encryptPassword(password,salt);
+            // Verify;
+            if (!passwordInDB.equals(codedPassword)) {
+              System.err.println("the password does not match");
```

Comment:
We don't want these error messages printed. Just type "//The password does not match" instead. Otherwise the user sees ugly command line errors -- the front end will take care of informing the user that the password failed using the boolean you return.